src/rebuild/srv.c

```diff
@@ -660,13 +629,27 @@ rebuild_leader_status_check(struct ds_pool *pool, uint32_t op,
 			dom = pool_map_find_node_by_rank(pool->sp_map, excluded.rl_ranks[i]);
 			D_ASSERT(dom != NULL);
 
-			/* If the rank is being scheduled after reintegration
-			 * start, let's skip the rank, i.e. set it DONE.
-			 */
-			if (dom->do_comp.co_status == PO_COMP_ST_UP &&
-			    dom->do_comp.co_in_ver < rgt->rgt_rebuild_ver)
-				continue;
-
+			if (rgt->rgt_opc == RB_OP_REBUILD) {
+				if (dom->do_comp.co_status == PO_COMP_ST_UP) {
+					if (dom->do_comp.co_in_ver > rgt->rgt_rebuild_ver) {
+						D_INFO(DF_UUID": cancel rebuild %u/%u\n",
+				       		       DP_UUID(pool->sp_uuid), rgt->rgt_rebuild_ver,
+						       dom->do_comp.co_in_ver);
+						rebuild_abort = true;
+						break;
+					} else {
+						continue;
+					}
+				} else if (dom->do_comp.co_status == PO_COMP_ST_DOWN) {
+					if (dom->do_comp.co_fseq > rgt->rgt_rebuild_ver) {
+						D_INFO(DF_UUID": cancel rebuild %u/%u\n",
+				       		       DP_UUID(pool->sp_uuid), rgt->rgt_rebuild_ver,
+						       dom->do_comp.co_fseq);
+						rebuild_abort = true;
```

Comment:
just confirm why started the rebuild (with the rgt) with "dom->do_comp.co_fseq > rgt->rgt_rebuild_ver" (and later abort it here)? the rebuild_prepare() should not start it?

Reply:
this means the pool map has been changed(co_fseq) during the rebuild, so let's abort the current rebuild and resolve them altogether in the next rebuild, since we have delay-rebuild, which has the ability to merge all operations in one rebuild.  Or I miss your question?